src/containers/CreatePipelineRun/CreatePipelineRun.js

```diff
@@ -90,10 +91,26 @@ const initialResourcesState = resourceSpecs => {
   return resourceSpecs.reduce(resourcesReducer, {});
 };
 
+// K8s label documentation comes from here:
+// https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+const labelKeyRegex = new RegExp(
+  '^(([a-z0-9A-Z]([a-z0-9A-Z-.]*[a-z0-9A-Z])?){0,253}/)?([a-z0-9A-Z]([a-z0-9A-Z-_.]*[a-z0-9A-Z])?){1,63}$'
+);
+const labelValueRegex = new RegExp(
+  '^([a-z0-9A-Z]([a-z0-9A-Z-_.]*[a-z0-9A-Z])?){0,63}$'
+);
+const validLabel = (type, value) => {
+  const regex = type === 'key' ? labelKeyRegex : labelValueRegex;
+  return regex.test(value);
+};
+
 class CreatePipelineRun extends React.Component {
   constructor(props) {
     super(props);
 
+    this.handleAddLabel = this.handleAddLabel.bind(this);
```

Comment:
You can avoid having to bind `this` to each handler by defining them as:

``` Javascript
handlerName = (arguments) => {}
```

Reply:
Oh, cool! I did not know this 👍